src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -161,6 +161,13 @@ private static void EmitRegexDerivedTypeRunnerFactory(IndentedTextWriter writer,
             writer.WriteLine($"        // Description:");
             DescribeExpression(writer, rm.Tree.Root.Child(0), "        // ", analysis); // skip implicit root capture
             writer.WriteLine();
+            if (rm.Tree.Culture != null)
+            {
+                // If the RegexTree has Culture set, then we need to emit the textInfo field that should be used at match time for casing operations.
+                writer.WriteLine($"        /// <summary>TextInfo that will be used for Backreference case comparisons.</summary>");
+                writer.WriteLine($"        private readonly TextInfo textInfo = CultureInfo.GetCultureInfo({Literal(rm.Tree.Culture.Name)}).TextInfo;");
```

Comment:
```suggestion
                writer.WriteLine($"        private readonly TextInfo _textInfo = CultureInfo.GetCultureInfo({Literal(rm.Tree.Culture.Name)}).TextInfo;");
```
and obviously update naming elsewhere accordingly.